Add edit profile and change password forms

tyanhan opened the pull request:
# Description
Added edit profile and change password forms on the profile page

## Type of change
- [X] New feature (non-breaking change which adds functionality)

# Screenshot
![profile](https://user-images.githubusercontent.com/68331979/147221926-3e5b99c9-1333-4be3-bdb0-3bb748487c1d.gif)


# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings

tyanhan committed:
Profile Page

tyanhan committed:
Add Profile Menu colour

MarcusTXK commented on frontend/components/molecules/MProfilePhoto.tsx:
```diff
@@ -0,0 +1,26 @@
+  return (
+    <Grid container direction="row" wrap="nowrap">
+      <Grid>
+        <Image src={profilePicture} alt="profile picture" />
```
Update the src to either use the avatar from the backend or to use dicebear's randomly generated profile picture based on their username.

src={avatar || `https://avatars.dicebear.com/api/bottts/${username}.svg`}

MarcusTXK commented on frontend/components/organisms/OChangePasswordForm.tsx:
```diff
@@ -0,0 +1,56 @@
+
+const OChangePasswordForm = () => {
+  const [currPass, setCurrPass] = useState('');
+  const [newPass, setNewPass] = useState('');
+  const [confirmPass, setConfirmPass] = useState('');
```
Add validation when they submit to check if these two are the same, if not warn the user. You can refer to `OLoginDialog`.

MarcusTXK commented on frontend/pages/profile.tsx:
```diff
@@ -0,0 +1,41 @@
+import OEditProfileForm from 'components/organisms/OEditProfileForm';
+import { useState } from 'react';
+
+const ProfilePage = () => {
+  const [page, setPage] = useState('edit_profile');
+
+  const handleClickEditProfile = () => {
+    setPage('edit_profile');
+  };
+
+  const handleClickChangePassword = () => {
+    setPage('change_password');
+  };
```
Consider using CamelCase for the values.

MarcusTXK reviewed (changes requested):
Wow, really good job! Looks really good.  Just some minor changes and it can be merged.

Some things to note:
Please use the thinner versions for the inputs (pass the prop `size="small"`) as the current ones are very thick. Also be mindful when editing shared components, such as changing their heights, as it breaks other components as seen here in the login modal, where the inputs are now of different sizes:
![image](https://user-images.githubusercontent.com/50147457/147316835-c558af67-6f2b-4c09-930e-b4a31503d56c.png)

Run `yarn lint` and `yarn build` and try to resolve all the errors
![image](https://user-images.githubusercontent.com/50147457/147317179-236342da-327f-4bae-b302-068c8c0762a9.png)

tyanhan committed:
Add Validation to change password

tyanhan commented:
I have added the validation to check if the new password is the same as the confirm password. I have also changed to input text boxes size to small and changed the picture such that it displays the image that was uploaded(or the randomly generated image if no file has been uploaded yet).

tyanhan committed:
Changed page names to CamelCase

MarcusTXK reviewed (approved):
LGTM, great job!
